---
title: "Fig2"
output: github_document
---


## Fig. 2: Heatmap of effect sizes for whole body tissue, cell type gene sets across traits (& Figure S2, S3, S13, S14, S17, S18)

#### -- Figure legend -- ####
Legend: 
Heatmap displays a representative set of the most significant associations between 5 categories of traits and deletions and duplications aggregated across (A) tissues and (B) whole body cell types. Traits are categorized and shown along the x-axis, while the y-axis lists gene sets, tissues, and cell types. The blue and red intensity color scale reflects the negative and positive effect sizes. Black asterisks indicate statistically significant associations between traits and genes (FDR correction across 172 ✕ 43 ✕ 2 =14,792 tests). C-F) Bar plots (with standard error) summarizing the differences in the level of association/enrichments between the type of variants for C, E) all gene-sets, and D, F) brain and non-brain functional gene sets. Black asterisks (*) indicate statistically significant proportion differences (FDR-adjusted). 
Abbreviations: HDL: high-density lipoprotein; HbA1c: glycated haemoglobin; BMD: bone mineral density; BMI: body mass index; EA: educational attainment; FI: fluid intelligence;  TDI: townsend deprivation index; VSWM: visuospatial working memory; FVC: Forced vital capacity; Del: deletion; Dup: duplication; SNP: single nucleotide polymorphism.


#### Libraries ####
```{r}

#### Libraries for Heatmap & Bar-Plots
library(ggplot2)
library(ggprism)
library(grid)
library(ggpubr)
library(gridExtra)
library(tidyr)
library(tibble)
library(dplyr)
library(ComplexHeatmap)
library(circlize)



```

#### FunBurd Associations' output : effectsize, se & pvalue####
```{r}
load('ST2.RData')
tail(ST2)
```


#### Whole Body Tissue Level (Fantom) - Heatmap of the effectsizes + FDR-pvalues  ####
```{r ,  fig.width=16, fig.height=15}

## Tissue Level - Fantom (43*60) 
tissue_order <- c("olfactory_bulb", "frontal_lobe", "medial_frontal_gyrus", "insular_cortex",
                  "paracentral_gyrus", "parietal_lobe", "temporal_cortex", "medial_temporal_gyrus", "postcentral_gyrus",
                  "hippocampus", "occipital_lobe", "occipital_cortex", "occipital_pole",
                  "amygdala", "caudate", "putamen", "substantia_nigra", "globus_pallidus", "thalamus",
                  "nucleus_accumbens", "pituitary_gland", "locus_coeruleus", "corpus_callosum",
                  "cerebellum", "pons", "medulla_oblongata", "spinal_cord", "retina",
                  "salivary_gland", "tongue", "esophagus", "thyroid_gland", "thymus",
                  "tonsil", "heart_muscle", "skeletal_muscle", "smooth_muscle", "lung", "liver", "gallbladder",
                  "spleen", "lymph_node", "pancreas", "adipose_tissue", "colon", "small_intestine",
                  "appendix", "kidney", "urinary_bladder",
                  "breast", "endometrium", "cervix", "vagina", "ovary", "placenta", "prostate", "seminal_vesicle",
                  "ductus_deferens", "epididymis", "testis")

trait_order <- c('Albumin', 'IGF1', 'Cholesterol', 'HDL', 'LDL',
                    'Calcium', 'Platelet', 'Redblood', 'Gamma', 'HbA1c', 'Triglycerides', 'Glucose',
                    'Menarche', 'Menopause', 'SexPartners', 'MET',
                    'FVC', 'FEV1', 'StandingHeight', 'BirthWeight', 'BMI', 'BMD', 'Hypertention', 'HeartRate',
                    'EA', 'FI', 'SymbolDigit', 'TrailMaking1', 'TrailMaking2', 'Reaction', 'VSWM1', 'VSWM2', 'TDI',
                    'RiskTaking', 'Depression', 'GP', 'Neuroticism', 'Irritability', 'MoodSwings', 'Miserableness', 'GuiltyFeelings', 'Loneliness','MoodAnxiety')


### Subset the data  #####
df_FantomTissue=subset(ST2,Geneset_Type=='Tissue_Fantom60')
df_FantomTissue_Del=subset(df_FantomTissue,CNV_Type=='DEL')
df_FantomTissue_Del_EffectSize=subset(df_FantomTissue_Del,select=c('Trait','Geneset','Effectsize'))
df_FantomTissue_Del_fdrpval=subset(df_FantomTissue_Del,select=c('Trait','Geneset','FDR_p_value'))
# Effectsize matrix
Matrix_FantomTissue_Del_EffectSize <- df_FantomTissue_Del_EffectSize %>%
  pivot_wider(names_from = Trait, values_from = Effectsize) %>%
  column_to_rownames("Geneset")

# FDR p-value matrix
Matrix_FantomTissue_Del_fdrpval <- df_FantomTissue_Del_fdrpval %>%
  pivot_wider(names_from = Trait, values_from = FDR_p_value) %>%
  column_to_rownames("Geneset")

# Order rows and columns for Del matrix
order_index <- match(rownames(Matrix_FantomTissue_Del_EffectSize), tissue_order)
Matrix_FantomTissue_Del_EffectSize_ordered <- Matrix_FantomTissue_Del_EffectSize[order(order_index), ]
Matrix_FantomTissue_Del_EffectSize_ordered <- Matrix_FantomTissue_Del_EffectSize_ordered[, match(trait_order, colnames(Matrix_FantomTissue_Del_EffectSize_ordered))]

# Order rows and columns for p-values
order_index <- match(rownames(Matrix_FantomTissue_Del_fdrpval), tissue_order)
Matrix_FantomTissue_Del_fdrpval_ordered <- Matrix_FantomTissue_Del_fdrpval[order(order_index), ]
Matrix_FantomTissue_Del_fdrpval_ordered <- Matrix_FantomTissue_Del_fdrpval_ordered[, match(trait_order, colnames(Matrix_FantomTissue_Del_fdrpval_ordered))]
Matrix_FantomTissue_Del_fdrpval_ordered <- ifelse(Matrix_FantomTissue_Del_fdrpval_ordered < 0.05, "*", "")

# View matrices
print(Matrix_FantomTissue_Del_EffectSize_ordered[1:5,40:43])


# Heatmap settings
colors <- colorRampPalette(c("#386cb0", "white", "#a50f15"))(100)
min_val <- -0.2
num_colors <- length(colors)
breaks <- seq(min_val, -min_val, length.out = num_colors)
color_function <- colorRamp2(breaks, colors)

# Create row annotations
row_annotation <- data.frame(
  Tissue = c(rep("Brain", 27), rep("Non-Brain", (60-27)))
)

row_annotation$Tissue <- factor(row_annotation$Tissue, levels = c("Brain", "Non-Brain"))

# Create column annotations
column_annotation <- data.frame(
  Trait = c(rep("Blood Assays",12),rep("Reproductive and\n Activity Factors",4),rep("Physical\n Measure",8),rep("Cognitive\n Metrics",9),rep("Mental\n Health",10))
)

column_annotation$Trait <- factor(column_annotation$Trait, levels = c("Blood Assays", "Reproductive and\n Activity Factors", "Physical\n Measure", "Cognitive\n Metrics", "Mental\n Health"))

# Define column heatmap annotation
column_ha <- HeatmapAnnotation(
  Trait = column_annotation$Trait,
  col = list(Trait = c("Blood Assays" = "#e78ac3", "Reproductive and\n Activity Factors" = "#8da0cb", "Physical\n Measure" = "#a6d854", "Cognitive\n Metrics" = "#66c2a5", "Mental\n Health" = "#fc8d62")),
  annotation_name = NULL,  # Remove both 'Supercluster' and 'Trait' labels
  show_legend = FALSE,
  show_annotation_name = FALSE
)

#png("Heatmap_DEL_Fantom_FDR.png",  width = 20000 , height = 25000,res = 1600 )
# Create the heatmap without dendrograms
ht1 <- Heatmap(
  as.matrix(Matrix_FantomTissue_Del_EffectSize_ordered), name = "Effect size",
  cluster_rows = FALSE,  # Disable row clustering
  cluster_columns = FALSE,  # Disable column clustering
  show_row_dend = FALSE,  # Do not show row dendrogram
  show_column_dend = FALSE,  # Do not show column dendrogram
  top_annotation = column_ha,
  row_split = row_annotation$Tissue,  # Split rows into 2 groups
  column_split = column_annotation$Trait,  # Split columns into 5 groups
  row_title = NULL,
  row_title_gp = gpar(fontsize = 8, fontface = "bold", fontfamily = "sans"),
  row_names_gp = gpar(col = "black", fontsize = 12, fontfamily = "sans", fontface = "bold"),
  column_title_gp = gpar(col = c('#e78ac3', '#8da0cb', '#a6d854', '#66c2a5', '#fc8d62'), fontsize = 14, fontface = "bold"),
  column_names_gp = gpar(col = "black", fontsize = 12, rot = 45, fontfamily = "sans", fontface = "bold"),
  column_names_rot = 45,
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(Matrix_FantomTissue_Del_fdrpval_ordered[i, j], x = x, y = y - convertHeight(grobHeight(textGrob('*')), "mm"), gp = gpar(fontsize = 36, col = 'grey40', fontfamily = "sans", just = "center", fontface = "bold",hjust=30))
  },
  col = color_function, row_gap = unit(0.5, "cm"),  # Adjust row margin (increase or decrease as needed)
  column_gap = unit(0.5, "cm"),
  show_heatmap_legend = FALSE,
  rect_gp = gpar(col = "lightgrey", lwd = 0.5)
)

#draw(ht1, padding = unit(c(0.5, 2, 0.5, 0.5), "cm"))
#dev.off()


# Draw the heatmap with padding
draw(ht1, padding = unit(c(0.5, 2, 0.5, 0.5), "cm"))



```

#### Whole Body Cell Type Level (HPA) - Heatmap of the effectsizes + FDR-pvalues ####
```{r ,  fig.width=16, fig.height=20}

## Cell Type Level - HPA (43*81) 
cell_order <- c("inhibitory_neurons", "excitatory_neurons", "astrocytes", "oligodendrocytes", "oligodendrocyte_precursor_cells", "microglial_cells",
                "muller_glia_cells", "melanocytes", "smooth_muscle_cells", "fibroblasts", "endometrial_stromal_cells", "mesothelial_cells",
                "endothelial_cells", "adipocytes", "lymphatic_endothelial_cells", "ovarian_stromal_cells", "granulosa_cells",
                "peritubular_cells", "leydig_cells", "sertoli_cells", "skeletal_myocytes", "cardiomyocytes", "t_cells", "nk_cells",
                "dendritic_cells", "plasma_cells", "b_cells", "erythroid_cells", "schwann_cells", "granulocytes", "monocytes",
                "kupffer_cells", "macrophages", "hofbauer_cells", "langerhans_cells", "proximal_tubular_cells", "collecting_duct_cells",
                "hepatocytes", "cholangiocytes", "exocrine_glandular_cells", "ductal_cells", "pancreatic_endocrine_cells", "distal_tubular_cells",
                "undifferentiated_cells", "intestinal_goblet_cells", "enteroendocrine_cells", "paneth_cells", "distal_enterocytes",
                "proximal_enterocytes", "gastric_mucus_secreting_cells", "prostatic_glandular_cells", "basal_prostatic_cells",
                "breast_myoepithelial_cells", "breast_glandular_cells", "salivary_duct_cells", "mucus_glandular_cells",
                "serous_glandular_cells", "alveolar_cells_type_2", "alveolar_cells_type_1", "secretory_cells", "glandular_and_luminal_cells",
                "ciliated_cells", "ionocytes", "basal_respiratory_cells", "club_cells", "squamous_epithelial_cells", "basal_squamous_epithelial_cells",
                "suprabasal_keratinocytes", "basal_keratinocytes", "syncytiotrophoblasts", "cytotrophoblasts", "extravillous_trophoblasts",
                "rod_photoreceptor_cells", "cone_photoreceptor_cells", "horizontal_cells", "bipolar_cells", "spermatogonia", "spermatocytes",
                "oocytes", "late_spermatids", "early_spermatids")


### Subset the data  #####
df_HPACell=subset(ST2,Geneset_Type=='SC_HPA81')
df_HPACell_Del=subset(df_HPACell,CNV_Type=='DEL')
df_HPACell_Del_EffectSize=subset(df_HPACell_Del,select=c('Trait','Geneset','Effectsize'))
df_HPACell_Del_fdrpval=subset(df_HPACell_Del,select=c('Trait','Geneset','FDR_p_value'))
# Effectsize matrix
Matrix_HPACell_Del_EffectSize <- df_HPACell_Del_EffectSize %>%
  pivot_wider(names_from = Trait, values_from = Effectsize) %>%
  column_to_rownames("Geneset")

# FDR p-value matrix
Matrix_HPACell_Del_fdrpval <- df_HPACell_Del_fdrpval %>%
  pivot_wider(names_from = Trait, values_from = FDR_p_value) %>%
  column_to_rownames("Geneset")

# Order rows and columns for Del matrix
order_index <- match(rownames(Matrix_HPACell_Del_EffectSize), cell_order)
Matrix_HPACell_Del_EffectSize_ordered <- Matrix_HPACell_Del_EffectSize[order(order_index), ]
Matrix_HPACell_Del_EffectSize_ordered <- Matrix_HPACell_Del_EffectSize_ordered[, match(trait_order, colnames(Matrix_HPACell_Del_EffectSize_ordered))]

# Order rows and columns for p-values
order_index <- match(rownames(Matrix_HPACell_Del_fdrpval), cell_order)
Matrix_HPACell_Del_fdrpval_ordered <- Matrix_HPACell_Del_fdrpval[order(order_index), ]
Matrix_HPACell_Del_fdrpval_ordered <- Matrix_HPACell_Del_fdrpval_ordered[, match(trait_order, colnames(Matrix_HPACell_Del_fdrpval_ordered))]
Matrix_HPACell_Del_fdrpval_ordered <- ifelse(Matrix_HPACell_Del_fdrpval_ordered < 0.05, "*", "")

# View matrices
print(Matrix_HPACell_Del_EffectSize_ordered[1:5,40:43])


# Create row annotations
row_annotation <- data.frame(
  Cell = c(rep("Brain",7 ), rep("Non-Brain", (81-7)))
)

row_annotation$Cell <- factor(row_annotation$Cell, levels = c("Brain", "Non-Brain"))

# all other setting are similar to the last chunk ###

ht2 <- Heatmap(
  as.matrix(Matrix_HPACell_Del_EffectSize_ordered), name = "Effect size",
  cluster_rows = FALSE,  # Disable row clustering
  cluster_columns = FALSE,  # Disable column clustering
  show_row_dend = FALSE,  # Do not show row dendrogram
  show_column_dend = FALSE,  # Do not show column dendrogram
  top_annotation = column_ha,
  row_split = row_annotation$Cell,  # Split rows into 2 groups
  column_split = column_annotation$Trait,  # Split columns into 5 groups
  row_title = NULL,
  row_title_gp = gpar(fontsize = 8, fontface = "bold", fontfamily = "sans"),
  row_names_gp = gpar(col = "black", fontsize = 12, fontfamily = "sans", fontface = "bold"),
  column_title_gp = gpar(col = c('#e78ac3', '#8da0cb', '#a6d854', '#66c2a5', '#fc8d62'), fontsize = 14, fontface = "bold"),
  column_names_gp = gpar(col = "black", fontsize = 12, rot = 45, fontfamily = "sans", fontface = "bold"),
  column_names_rot = 45,
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(Matrix_HPACell_Del_fdrpval_ordered[i, j], x = x, y = y - convertHeight(grobHeight(textGrob('*')), "mm"), gp = gpar(fontsize = 36, col = 'grey40', fontfamily = "sans", just = "center", fontface = "bold",hjust=30))
  },
  col = color_function, row_gap = unit(0.5, "cm"),  # Adjust row margin (increase or decrease as needed)
  column_gap = unit(0.5, "cm"),
  show_heatmap_legend = FALSE,
  rect_gp = gpar(col = "lightgrey", lwd = 0.5)
)
# Draw the heatmap with padding
draw(ht2, padding = unit(c(0.5, 2, 0.5, 0.5), "cm"))
```


#### Whole Brain Cell Type Level (Siletti) - Heatmap of the effectsizes + FDR-pvalues ####

```{r ,  fig.width=16, fig.height=10}
braincell_order = c(
     "amygdala_excitatory","cerebellar_inhibitory","cge_interneuron",               
     "deep_layer_corticothalamic_and_6b" ,"deep_layer_intratelencephalic","deep_layer_near_projecting",     
     "eccentric_medium_spiny_neuron","hippocampal_ca1_3","hippocampal_ca4", "hippocampal_dentate_gyrus",
     "lamp5_lhx6_and_chandelier","lower_rhombic_lip", "mammillary_body","medium_spiny_neuron","mge_interneuron",             
     "midbrain_derived_inhibitory","miscellaneous","splatter","thalamic_excitatory","upper_layer_intratelencephalic",
     "upper_rhombic_lip","astrocyte", "bergmann_glia", "choroid_plexus","committed_oligodendrocyte_precursor", "ependymal",
     "fibroblast","microglia","oligodendrocyte", "oligodendrocyte_precursor", "vascular"
  )

### Subset the data  #####
df_SilettiCell=subset(ST2,Geneset_Type=='SC_Siletti31')
df_SilettiCell_Del=subset(df_SilettiCell,CNV_Type=='DEL')
df_SilettiCell_Del_EffectSize=subset(df_SilettiCell_Del,select=c('Trait','Geneset','Effectsize'))
df_SilettiCell_Del_fdrpval=subset(df_SilettiCell_Del,select=c('Trait','Geneset','FDR_p_value'))
# Effectsize matrix
Matrix_SilettiCell_Del_EffectSize <- df_SilettiCell_Del_EffectSize %>%
  pivot_wider(names_from = Trait, values_from = Effectsize) %>%
  column_to_rownames("Geneset")

# FDR p-value matrix
Matrix_SilettiCell_Del_fdrpval <- df_SilettiCell_Del_fdrpval %>%
  pivot_wider(names_from = Trait, values_from = FDR_p_value) %>%
  column_to_rownames("Geneset")


# Order rows and columns for Del matrix
order_index <- match(rownames(Matrix_SilettiCell_Del_EffectSize), braincell_order)
Matrix_SilettiCell_Del_EffectSize_ordered <- Matrix_SilettiCell_Del_EffectSize[order(order_index), ]
Matrix_SilettiCell_Del_EffectSize_ordered <- Matrix_SilettiCell_Del_EffectSize_ordered[, match(trait_order, colnames(Matrix_SilettiCell_Del_EffectSize_ordered))]

# Order rows and columns for p-values
order_index <- match(rownames(Matrix_SilettiCell_Del_fdrpval), braincell_order)
Matrix_SilettiCell_Del_fdrpval_ordered <- Matrix_SilettiCell_Del_fdrpval[order(order_index), ]
Matrix_SilettiCell_Del_fdrpval_ordered <- Matrix_SilettiCell_Del_fdrpval_ordered[, match(trait_order, colnames(Matrix_SilettiCell_Del_fdrpval_ordered))]
Matrix_SilettiCell_Del_fdrpval_ordered <- ifelse(Matrix_SilettiCell_Del_fdrpval_ordered < 0.05, "*", "")

# View matrices
print(Matrix_SilettiCell_Del_EffectSize_ordered[1:5,40:43])

row_annotation <- data.frame(
  braincell = c(rep("Neuron", 21), rep("Non-Neuron", 10))
)

row_annotation$braincell <- factor(row_annotation$braincell, levels = c("Neuron", "Non-Neuron"))

# all other setting are similar to the last chunk ###

ht3 <- Heatmap(
  as.matrix(Matrix_SilettiCell_Del_EffectSize_ordered), name = "Effect size",
  cluster_rows = FALSE,  # Disable row clustering
  cluster_columns = FALSE,  # Disable column clustering
  show_row_dend = FALSE,  # Do not show row dendrogram
  show_column_dend = FALSE,  # Do not show column dendrogram
  top_annotation = column_ha,
  row_split = row_annotation$braincell,  # Split rows into 2 groups
  column_split = column_annotation$Trait,  # Split columns into 5 groups
  row_title = NULL,
  row_title_gp = gpar(fontsize = 8, fontface = "bold", fontfamily = "sans"),
  row_names_gp = gpar(col = "black", fontsize = 12, fontfamily = "sans", fontface = "bold"),
  column_title_gp = gpar(col = c('#e78ac3', '#8da0cb', '#a6d854', '#66c2a5', '#fc8d62'), fontsize = 14, fontface = "bold"),
  column_names_gp = gpar(col = "black", fontsize = 12, rot = 45, fontfamily = "sans", fontface = "bold"),
  column_names_rot = 45,
  cell_fun = function(j, i, x, y, width, height, fill) {
    grid.text(Matrix_SilettiCell_Del_fdrpval_ordered[i, j], x = x, y = y - convertHeight(grobHeight(textGrob('*')), "mm"), gp = gpar(fontsize = 36, col = 'grey40', fontfamily = "sans", just = "center", fontface = "bold",hjust=30))
  },
  col = color_function, row_gap = unit(0.5, "cm"),  # Adjust row margin (increase or decrease as needed)
  column_gap = unit(0.5, "cm"),
  show_heatmap_legend = FALSE,
  rect_gp = gpar(col = "lightgrey", lwd = 0.5)
)
# Draw the heatmap with padding
draw(ht3, padding = unit(c(0.5, 2, 0.5, 0.5), "cm"))

```

#### For Dup the same script is used. ####



#### Global Proportion Test For Del, Dup & GWAS(S-LDSC) ####

```{r , fig.width=8, fig.height=4}
load('ST8.RData')
head(ST8)

# Reshape data: gather proportions for both Type1 and Type2
Global_Prop <- ST8 %>%
  pivot_longer(cols = starts_with("estimate_prop"),
               names_to = "Type_source", values_to = "prop") %>%
  mutate(Type = ifelse(Type_source == "estimate_prop_Type1", Type1, Type2)) %>%
  select(Geneset_Type, Type, prop) %>%
  distinct()   # avoid duplicates

# Add n depending on Geneset_Type (for computing se of the proportion)
Global_Prop <- Global_Prop %>%
  mutate(n = case_when(
    Geneset_Type == "Fantom"  ~ 60*43,
    Geneset_Type == "HPA"     ~ 81*43,
    Geneset_Type == "Siletti" ~ 31*43
  )) %>%
  mutate(se = sqrt(prop * (1 - prop) / n))

# Custom colors
custom_colors <- c("SNP" = "darkgreen", "Del" = "red3", "Dup" = "royalblue")
# Plot
ggplot(Global_Prop, aes(x = Type, y = prop, fill = Type)) +
  geom_bar(stat = "identity", width = 0.4) +
  geom_errorbar(aes(ymin = prop - se, ymax = prop + se), width = 0.15) +
  scale_fill_manual(values = custom_colors) +
  labs(title = " ",
       y = "Proportion", x = " ") +
  ylim(0, 0.25) +
  facet_wrap(~Geneset_Type, nrow = 1) +
  theme_prism(base_size = 14) +
  theme(legend.position = "none",
        axis.line = element_line(linewidth = 0.8),
        axis.text.y = element_text(size = 16))

print(subset(ST8,select=c('Geneset_Type', 'Type1', 'Type2', 'p_value_fdr')))


```


#### Brain-Nbrain Proportion Test For Del, Dup & GWAS(S-LDSC) ####
```{r,  fig.width=6, fig.height=10}
load('ST9.RData')
head(ST9)



# Prepare data
BrainNbrainNeuronNonNeuron_Prop <- ST9 %>%
  pivot_longer(cols = starts_with("estimate_prop"),
               names_to = "BrainNbrain", values_to = "prop") %>%
  mutate(
    # Assign Brain/NonBrain labels
    BrainNbrain = ifelse(BrainNbrain == "estimate_prop_Brain", "Brain", "NonBrain")
  )

# Add n and compute SE
BrainNbrainNeuronNonNeuron_Prop <- BrainNbrainNeuronNonNeuron_Prop %>%
  mutate(n = case_when(
    Geneset_Type == "Fantom"  ~ 60*43,
    Geneset_Type == "HPA"     ~ 81*43,
    Geneset_Type == "Siletti" ~ 31*43
  )) %>%
  mutate(se = sqrt(prop * (1 - prop) / n))

# Custom colors
custom_colors <- c("Brain" = "#d95f02",
                   "NonBrain" = "#66a61e",
                   "Neuron" = "#d95f02",
                   "NonNeuron" = "#66a61e")

# --- Plot Fantom + HPA ---
BrainNbrain_Prop <- BrainNbrainNeuronNonNeuron_Prop %>% filter(Geneset_Type %in% c("Fantom", "HPA"))
ggplot(BrainNbrain_Prop, aes(x = BrainNbrain, y = prop, fill = BrainNbrain)) +
  geom_bar(stat = "identity", width = 0.4) +
  geom_errorbar(aes(ymin = prop - se, ymax = prop + se), width = 0.1) +
  scale_fill_manual(values = custom_colors) +
  labs(title = " ", y = "Proportion", x = "") +
  ylim(0, 0.4) +
  facet_grid(Type ~ Geneset_Type) +
  theme_prism(base_size = 14) +
  theme(
    legend.position = "none",
    axis.line = element_line(linewidth  = 0.8),
    axis.line.x = element_line(size = 0.8),
    axis.text.y = element_text(size = 16)
  )

# --- Plot Siletti ---
NeuronNonNeuron_Prop <- BrainNbrainNeuronNonNeuron_Prop %>%
  filter(Geneset_Type == "Siletti") %>%
  mutate(NeuronNonNeuron = ifelse(BrainNbrain == "Brain", "Neuron", "NonNeuron"))

ggplot(NeuronNonNeuron_Prop, aes(x = NeuronNonNeuron, y = prop, fill = NeuronNonNeuron)) +
  geom_bar(stat = "identity", width = 0.2) +
  geom_errorbar(aes(ymin = prop - se, ymax = prop + se), width = 0.05) +
  scale_fill_manual(values = custom_colors) +
  labs(title = " ", y = "Proportion", x = " ") +
  ylim(0, 0.4) +
  facet_grid(Type ~ Geneset_Type) +
  theme_prism(base_size = 14) +
  theme(
    legend.position = "none",
    axis.line = element_line(linewidth = 0.8),
    axis.line.x = element_line(size = 0.8),
    axis.text.y = element_text(size = 16)
  )
print(subset(ST9,select=c('Geneset_Type', 'Type', 'p_value_fdr')))


```

