---
title: "Fig4"
output: github_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE)
```


## Fig. 4: Rare and common variant architectures across complex traits.##

## Markdown for generating panel figures and statistics ##

## --------- Figure legend --------- ###
Legend: 
A) Example of genetic and phenotypic correlations between fluid intelligence (FI) and 6 traits, with each point color-coded by correlation type: SNP-based genetic correlations, deletion, duplication burden correlations, loss-of-function SNV burden correlation, and phenotypic correlations.  B) The heatmap of burden correlations, for deletions and duplications in the lower and upper triangle matrices, respectively. Significant correlations are marked by asterisks (FDR corrected p-Jaccard). C) The scatter plot compares effect size correlations between deletions and duplications, where orange lines represent the Total Least Squares regression. The dashed line indicates perfect concordance (y = x). We provide the same comparison between CNVs and (D) SNP-based genetic correlations and (E) SNV burden correlations; phenotypic correlations are shown in Figure S20. F) Boxplots illustrate the variance explained by gene sets for both brain and non-brain traits, based on regression analysis.
Abbreviations: BMI: body mass index; EA: educational attainment; FI: fluid intelligence;  TDI: townsend deprivation index; VSWM: visuospatial working memory; FVC: Forced vital capacity; Del: deletion; Dup: duplication; SNP: single nucleotide polymorphism; SNV: single nucleotide variants; LoF: loss of function.

#### Libraries for Heatmap, ScatterPlot & ForestPlot
```{r}
library(ggplot2)
library(ggcorrplot)
library(ggprism)
library(ggpubr)
library(dplyr)
library(tidyr)
library(reshape2)
library(Rmpfr)
library(MASS)
library(vegan)
library(DescTools)
library(missForest)
library(glue)
library(scales)
library(forcats)
library(RColorBrewer)
```
#### Read pairwise correlations dataset based on Del,Dup,CNV, SNV and SNP

```{r}

## EffectSize Correlation / CNV and Del and Dup Seperately

PathData="/Users/sayekazem/Desktop/New_Project_2023_Summer/PaperMultitrait2024_Data_FinalVersion/"
PathFig="/Users/sayekazem/Desktop/New_Project_2023_Summer/PaperMultitrait2024_Figs_FinalVersion/"

#### Pheno Corr ############################
Pheno_Corr=read.csv(paste0(PathData,'Pairwise_GeneticCorrelations/ALL_Pairwise_Corr_Pheno_File_AllTraits_Feb2025.csv'))

#### SNP Corr ##############################
GWAS_Corr=read.csv(paste0(PathData,'Pairwise_GeneticCorrelations/ALL_Pairwise_Corr_MixNealeWiener_Common_File_37Traits_Feb2025.csv'))

#### SNV Corr ##############################
Corr_SNV=read.csv(paste0(PathData,'Pairwise_GeneticCorrelations/ALL_Pairwise_Corr_Wiener_SNVs_23Traits_Feb2025.csv'))

#### CNV Corr ##############################
CNVs_Corr=read.csv(paste0(PathData,'Pairwise_GeneticCorrelations/ALL_Pairwise_Corr_CNVs_File_AllTraits_Feb2025.csv'))

#### DEL/Dup Corr ##############################
DELDUP_Corr=read.csv(paste0(PathData,'Pairwise_GeneticCorrelations/ALL_Pairwise_Corr_DELDUP_File_AllTraits_Feb2025.csv'))

DEL_Corr=subset(DELDUP_Corr,select = c('Trait1','Trait2','Correlation_DEL'))
DUP_Corr=subset(DELDUP_Corr,select = c('Trait1','Trait2','Correlation_DUP'))

head(DELDUP_Corr)
```
#
#### Preparing data for the heatmap plot : Matrix format of pairwise corrrelation + Jaccard-based pvalues for subset of traits for CNVs (Del & Dup)

```{r}

### Matrix format of pairwise correlations  ########################################################
DEL_Corr_Mat=read.csv(paste0(PathData,'Pairwise_GeneticCorrelations/Matrix_ALL_Pairwise_Corr_DEL_File_AllTraits_Feb2025.csv'))
DUP_Corr_Mat=read.csv(paste0(PathData,'Pairwise_GeneticCorrelations/Matrix_ALL_Pairwise_Corr_DUP_File_AllTraits_Feb2025.csv'))
#
rownames(DEL_Corr_Mat) <- DEL_Corr_Mat$X
DEL_Corr_Mat$X <- NULL 

rownames(DUP_Corr_Mat) <- DUP_Corr_Mat$X
DUP_Corr_Mat$X <- NULL 
#Subset of traits:
Subset_traits=c('Loneliness','MoodAnxiety','Neuroticism','Depression..self.report.','TDI','EA','FI_merged','VSWM1','Platelet','Redblood','HeartRate','BMI','FVC','Menarche')
Corre_File_DUP_subset=subset(DUP_Corr_Mat,select=Subset_traits)

Corre_File_DUP_subset=DUP_Corr_Mat[Subset_traits,]

Corre_File_DEL_subset=subset(DEL_Corr_Mat,select=Subset_traits)
Corre_File_DEL_subset=DEL_Corr_Mat[Subset_traits,]

#Predefined order
new_order <- c("Platelet" , "Redblood" ,'VSWM1','BMI','TDI',
               'HeartRate','Loneliness',"Neuroticism",'MoodAnxiety',"Depression..self.report.","FI_merged",
               'EA',"FVC" ,"Menarche")
new_name = c("Platelet" , "Redblood" ,'VSWM1','BMI','TDI',
             'HeartRate','Loneliness',"Neuroticism","MoodAnxiety","Depression","FI",
             'EA',"FVC" ,"Menarche")

Corre_File_DEL_subset=Corre_File_DEL_subset[new_order,new_order]
Corre_File_DUP_subset=Corre_File_DUP_subset[new_order,new_order]

rownames(Corre_File_DEL_subset)=new_name
colnames(Corre_File_DEL_subset)=new_name

rownames(Corre_File_DUP_subset)=new_name
colnames(Corre_File_DUP_subset)=new_name
############################################################

combined_corr_subset <- Corre_File_DEL_subset
combined_corr_subset[upper.tri(combined_corr_subset)] <- Corre_File_DUP_subset[upper.tri(Corre_File_DUP_subset)]
diag(combined_corr_subset)=Inf


### Read Jaccard-Based pval ########################################################
Pval_Jaccard=read.csv('/Users/sayekazem/Desktop/New_Project_2023_Summer/PaperMultitrait2024_Data_FinalVersion/BrainSmash/Pairwise_Correlations/Surrogate_Pvalues_Stacked_Pairwise_DelDup_Feb2025.csv')
Pval_Jaccard <- Pval_Jaccard %>%
  mutate(trait1 = case_when(
    trait1 == "Depression (self-report)" ~ "Depression..self.report.",
    trait1 == "Mood Swings" ~ "Mood.Swings",
    trait1 == "Risk Taking" ~ "Risk.Taking",
    trait1=='Seen a GP'~ "Seen.a.GP",
    TRUE ~ trait1  # Keep other names unchanged
  ))
Pval_Jaccard <- Pval_Jaccard %>%
  mutate(trait2 = case_when(
    trait2 == "Depression (self-report)" ~ "Depression..self.report.",
    trait2 == "Mood Swings" ~ "Mood.Swings",
    trait2 == "Risk Taking" ~ "Risk.Taking",
    trait2=='Seen a GP'~ "Seen.a.GP",
    
    TRUE ~ trait2  # Keep other names unchanged
  ))

Pval_Jaccard=Pval_Jaccard[,c(2:length(Pval_Jaccard))]

#### Subset of Jaccard p-values #################################################

Pval_Jaccard_subset <- subset(Pval_Jaccard, 
                              trait1 %in% Subset_traits & 
                                trait2 %in% Subset_traits)

## Create pval matrix DEL
pval_matrix_DEL <- matrix(NA, nrow = length(new_order), ncol = length(new_order),
                          dimnames = list(new_order, new_order))

# Fill the matrix with surrogate_p_value_stacked_DEL values
for (i in 1:nrow(Pval_Jaccard_subset)) {
  trait1 <- Pval_Jaccard_subset$trait1[i]
  trait2 <- Pval_Jaccard_subset$trait2[i]
  
  value1 <- Pval_Jaccard_subset$surrogate_p_value_stacked_DEL[
    (Pval_Jaccard_subset$trait1 == trait1 & Pval_Jaccard_subset$trait2 == trait2)
  ]
  value2 <- Pval_Jaccard_subset$surrogate_p_value_stacked_DEL[
    (Pval_Jaccard_subset$trait1 == trait2 & Pval_Jaccard_subset$trait2 == trait1)
  ]
  
  value=min(value1, value2, na.rm = TRUE)
  
  pval_matrix_DEL[trait1, trait2] <- value
  pval_matrix_DEL[trait2, trait1] <- value
  
}

## Create pval matrix DUP
pval_matrix_DUP <- matrix(NA, nrow = length(new_order), ncol = length(new_order),
                          dimnames = list(new_order, new_order))

# Fill the matrix with surrogate_p_value_stacked_DEL values
for (i in 1:nrow(Pval_Jaccard_subset)) {
  trait1 <- Pval_Jaccard_subset$trait1[i]
  trait2 <- Pval_Jaccard_subset$trait2[i]
  value <- Pval_Jaccard_subset$surrogate_p_value_stacked_DUP[i]
  pval_matrix_DUP[trait1, trait2] <- value
  pval_matrix_DUP[trait2, trait1] <- value
}

rownames(pval_matrix_DUP)=new_name
colnames(pval_matrix_DUP)=new_name

rownames(pval_matrix_DEL)=new_name
colnames(pval_matrix_DEL)=new_name


###################
combined_corr_Jacpvals_subset <- pval_matrix_DEL
combined_corr_Jacpvals_subset[upper.tri(combined_corr_Jacpvals_subset)] <- pval_matrix_DUP[upper.tri(pval_matrix_DUP)]
#combined_corr_Jacpvals=1-combined_corr_Jacpvals
diag(combined_corr_Jacpvals_subset)=Inf

# Convert p-values to stars manually
significance_levels <- function(p) {
  ifelse(p < 0.05, "*", "")
}

stars_matrix <- matrix(significance_levels(combined_corr_Jacpvals_subset), 
                       nrow = nrow(combined_corr_Jacpvals_subset))
rownames(stars_matrix) <- rownames(combined_corr_Jacpvals_subset)
colnames(stars_matrix) <- colnames(combined_corr_Jacpvals_subset)


########
```
#### Panel B of Fig 3 : Heatmap plot of pairwise effectsize correlation + pvalues for subset of traits (Del(lower triangle matrix) and Dup (upper triangle matrix))

```{r}
#png(paste0(PathFig,"Main_Figs/CorrelationPlot_Spearman_DELDUP_SubsetTraits_TissueCellSilletiSuper_Feb2025.png"), width = 5000, height = 5000, res = 700)
# Generate the correlation plot
p <- ggcorrplot(combined_corr_subset, 
                type = "full", 
                colors = c("darkblue", "white", "darkred")) 
p <- p+ theme_prism(axis_text_angle = 45,border = FALSE,base_size = 14,
                    base_family = "sans",
                    base_fontface = "bold") 

# Overlay manually positioned stars
for (i in 1:nrow(combined_corr_subset)) {
  for (j in 1:ncol(combined_corr_subset)) {
    if (stars_matrix[i, j] != "") { # Only add stars if significant
      p <- p + annotate("text", x = i, y=j-0.2, label = stars_matrix[i, j], 
                        color = "grey25", size = 10, fontface = "bold")
    }
  }
}

print(p)

```


#### Panel B of Fig 3 : Scatter plot of pairwise effectsize correlation for all pairs of traits

```{r}

### Function for TLS slope of the scatterplot):

# Total Least Squares function
total_least_squares <- function(X, Y) {
  X <- as.matrix(X)
  Y <- as.matrix(Y)
  Z <- cbind(X, Y)
  svd_result <- svd(Z)
  U <- svd_result$u
  S <- svd_result$d
  V <- svd_result$v
  nx <- ncol(X)
  Vxy <- V[1:nx, (nx + 1):ncol(V)]
  Vyy <- V[(nx + 1):ncol(V), (nx + 1):ncol(V)]
  B_tls <- -Vxy %*% solve(Vyy)
  return(B_tls)
}


compute_tls_p_value <- function(X, Y, slope) {
  n <- length(X)
  if (n <= 2) {
    stop("Not enough data points to compute p-value.")
  }
  
  mean_x <- mean(X)
  mean_y <- mean(Y)
  s_xx <- sum((X - mean_x)^2)
  s_yy <- sum((Y - mean_y)^2)
  s_xy <- sum((X - mean_x) * (Y - mean_y))
  
  if (s_xx == 0 || s_yy == 0) {
    stop("Variance of X or Y is zero.")
  }
  
  std_err <- sqrt((s_yy - slope * s_xy) / (n - 2)) / sqrt(s_xx)
  t_statistic <- slope / std_err
  
  # Use Rmpfr for higher precision t-statistic
  t_statistic_mpfr <- mpfr(t_statistic, precBits = 256)
  
  # Calculate the p-value manually using the CDF of the t-distribution
  df <- n - 2
  p_value <- 2 * (1 - as.numeric(pf(as.numeric(t_statistic_mpfr)^2, df1 = 1, df2 = df)))
  
  return(p_value)
}

Y <- DELDUP_Corr$Correlation_DEL
X <- DELDUP_Corr$Correlation_DUP

# Calculate the TLS slope
tls_slope <- total_least_squares(X, Y)
tls_p=compute_tls_p_value(X,Y,tls_slope)

##################################################

# Add the TLS line to the ggplot
#png("/Users/sayekazem/Desktop/New_Project_2023_Summer/PaperMultitrait2024_Figs_FinalVersion/Main_Figs/CorrelationPlot_DUPDEL_Spearman_43Traits_TLS_All_Stacked_Feb2025.png", width = 1500, height = 1500, res = 350)

ggplot(DELDUP_Corr, aes(x = Correlation_DUP, y = Correlation_DEL)) + 
  geom_point(color = "grey" ,alpha=0.4) +
  labs(
    y = "Effect Size Correlation - Del",
    x = "Effect Size Correlation - Dup"
  ) +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "darkgrey") +
  geom_abline(slope = tls_slope, intercept = 0, color = "orange",size=1) + # TLS line
  coord_cartesian(ylim = c(-1, +1), xlim = c(-1, +1)) +
  theme_prism(border = FALSE,base_size = 14,
              base_family = "sans",
              base_fontface = "bold") +
  theme(axis.line = element_line(size = 0.8))     # Change size of y-axis labels

print(tls_slope)
```
#### Panel C (Deletion and SNP)
```{r}
# Merge CNV and GWAS : Use the mutual traits between CNV and GWAS
DELDUPSNP_Corr_Subset = merge(DELDUP_Corr, GWAS_Corr, by.x=c('Trait1', 'Trait2'), by.y=c('Trait1', 'Trait2'))

Y <- DELDUPSNP_Corr_Subset$Correlation_DEL
X <- DELDUPSNP_Corr_Subset$Common_Correlation

# Calculate the TLS slope
tls_slope <- total_least_squares(X, Y)
#tls_p=compute_tls_p_value(X,Y,tls_slope)

ggplot(DELDUPSNP_Corr_Subset, aes(x = Common_Correlation, y = Correlation_DEL)) + 
  geom_point(color = "grey" ,alpha=0.4) +
  labs(
    y = "Effect Size Correlation - Del",
    x = "Genetic Correlation - SNP"
  ) +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "darkgrey") +
  geom_abline(slope = tls_slope, intercept = 0, color = "orange",size=1) + # TLS line
  coord_cartesian(ylim = c(-1, +1), xlim = c(-1, +1)) +
  theme_prism(border = FALSE,base_size = 14,
              base_family = "sans",
              base_fontface = "bold") +
  theme(axis.line = element_line(size = 0.8))     # Change size of y-axis labels

print(tls_slope)
```
#### Panel D (Deletion and SNV)
```{r}
# Merge CNV and SNV : Use the mutual traits between CNV and SNV
DELDUPSNV_Corr_Subset = merge(DELDUP_Corr, Corr_SNV, by.x=c('Trait1', 'Trait2'), by.y=c('Trait1', 'Trait2'))

Y <- DELDUPSNV_Corr_Subset$Correlation_DEL
X <- DELDUPSNV_Corr_Subset$SNVs_Correlation_Wiener

# Calculate the TLS slope
tls_slope <- total_least_squares(X, Y)
#tls_p=compute_tls_p_value(X,Y,tls_slope)

ggplot(DELDUPSNV_Corr_Subset, aes(x = SNVs_Correlation_Wiener, y = Correlation_DEL)) + 
  geom_point(color = "grey" ,alpha=0.4) +
  labs(
    y = "Effect Size Correlation - Del",
    x = "Burden Genetic Correlation - SNV"
  ) +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "darkgrey") +
  geom_abline(slope = tls_slope, intercept = 0, color = "orange",size=1) + # TLS line
  coord_cartesian(ylim = c(-1, +1), xlim = c(-1, +1)) +
  theme_prism(border = FALSE,base_size = 14,
              base_family = "sans",
              base_fontface = "bold") +
  theme(axis.line = element_line(size = 0.8))     # Change size of y-axis labels

print(tls_slope)
```

#### Panel A (genetic and phenotypic correlations between FI and a subset of traits)

```{r}
## Creating the full matrix
PhenSNP=merge(Pheno_Corr,GWAS_Corr,by.x = c('Trait1','Trait2'),by.y=c('Trait1','Trait2'))
CNVSNV=merge(DELDUP_Corr,Corr_SNV,by.x = c('Trait1','Trait2'),by.y=c('Trait1','Trait2'))



CNVSNV$key <- apply(CNVSNV[, c("Trait1", "Trait2")], 1, function(x) paste(sort(x), collapse = "_"))
PhenSNP$key <- apply(PhenSNP[, c("Trait1", "Trait2")], 1, function(x) paste(sort(x), collapse = "_"))
merge_all <- merge(CNVSNV, PhenSNP, by = "key", all = FALSE)
merge_all <- subset(merge_all, select = c("Trait1.x", "Trait2.x", "Correlation_DEL", "Correlation_DUP", 
                                          "SNVs_Correlation_Wiener", "Correlation_Pheno", "Common_Correlation"))
merge_all <- merge_all %>%
  rename(Trait1 = Trait1.x, Trait2 = Trait2.x)

##########################################
## Preparing for Plot ###

FI_Others <- merge_all %>%
  filter(Trait1 == "FI_merged" | Trait2 == "FI_merged")#
# Edit TDI
FI_Others$Trait1='FI_merged'
FI_Others <- FI_Others %>%
  mutate(Trait2 = ifelse(Trait2 == "FI_merged", "TDI", Trait2))

# Reshape the data to long format
long_data <- FI_Others %>%
  pivot_longer(cols = c(Correlation_DEL,Correlation_DUP, SNVs_Correlation_Wiener, Correlation_Pheno, Common_Correlation),
               names_to = "correlation_type",
               values_to = "correlation") %>%
  mutate(correlation_type = case_when(
    correlation_type == "Correlation_DEL" ~ "DEL-burden",
    correlation_type == "Correlation_DUP" ~ "DUP-burden",
    correlation_type == "SNVs_Correlation_Wiener" ~ "SNV-lof-burden",
    correlation_type == "Common_Correlation" ~ "SNP (rg)",
    correlation_type == "Correlation_Pheno" ~ "Phenotype",
    TRUE ~ correlation_type
  ))

FI_Others_plot=long_data

### Reordering Data ######
ordered_trait2 <- FI_Others_plot %>%
  #filter(correlation_type == "SNV-lof-burden") %>%
  filter(correlation_type == "DEL-burden") %>%
  arrange(correlation) %>%
  pull(Trait2)

custom_order <- c( "DEL-burden","DUP-burden","SNV-lof-burden", "SNP (rg)","Phenotype")

FI_Others_plot <- FI_Others_plot %>%
  mutate(
    Trait2 = factor(Trait2, levels = ordered_trait2),
    correlation_type = factor(correlation_type, levels = custom_order)
  )

# Order the data frame for plotting
FI_Others_plot <- FI_Others_plot %>%
  arrange(Trait2, correlation_type)


FI_Others_plot=subset(FI_Others_plot, Trait2 %in% c("Reaction","Neuroticism","TDI","Redblood","BMI","StandingHeight"))
FI_Others_plot <- FI_Others_plot %>%
  mutate(Trait2 = fct_recode(Trait2, "Height" = "StandingHeight"))

### Plot
custom_colors <- c(
  "DEL-burden" = "red3", 
  "DUP-burden" = "royalblue",# Set to your preferred colors
  "SNV-lof-burden" = "orange3",
  "SNP (rg)" = "green4",
  "Phenotype" = "purple4"
)
colors <- custom_colors[unique(FI_Others_plot$correlation_type)]
# Plot
ggplot(FI_Others_plot, aes(x=correlation, y=Trait2, color=correlation_type)) +
  geom_errorbarh(aes(xmin = 0, xmax = correlation),
                 position = position_dodge(width = 0.5),height = 0,alpha=0.7) +
  geom_point(size=2,aes(x=correlation, y=Trait2, color=correlation_type),
             position = position_dodge(width = 0.5),alpha=0.8) +
  geom_vline(lty = 2, xintercept = 0,color='darkgrey') +
  #scale_x_continuous(labels = scales::percent_format()) +
  #scale_color_manual(values=rainbow(4)) + 
  scale_color_manual(values=colors)+# Adjust colors for correlation_type
  #scale_size_continuous(labels = scales::comma) +
  
  labs(x = "Correlation with FI",
       y = "",
       color = "Correlation Type")+
  #size = "Correlation Value") +
  #theme_light() 
  theme_prism()+
  #coord_flip() +
  theme(
    panel.grid.major.y = element_blank(),
    panel.border = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.y = element_text(size = 12),axis.line = element_line(size = 0.8)) 

```